registry/registry.go

```diff
@@ -16,6 +15,12 @@ import (
 
 type Option int
 
+// LogPrinter hides the logging function Printf behind a simple
+// interface so libraries such as logrus can be used.
+type LogPrinter interface {
```

Comment:
Could we change this interface name to just be `Logger`, and also define it in both package `grid` and `registry`.

The `registry` package should not be used by users, its just internals.

Otherwise the changes look good.